src/include/common/vector/auxiliary_buffer.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "arrow/array.h"
```

Comment:
Let's think of a way to get rid of this include. I hope we can go through `common/arrow`. One way is to have a zero-copy conversion in copier from arrow's arrow_array to common's arrow_array.